doc/contribute/modifying_contributions.rst

```diff
@@ -0,0 +1,67 @@
+.. _modifying_contributions:
+
+Modifying Contributions made by other developers
+************************************************
+
+Scenarios
+#########
+
+In general, Zephyr contributors and collaborators are encouraged to assist
+as reviewers in pull requests, so that patches may be approved and merged
+to Zephyr's main branch as part of the original pull requests. The authors
+of the pull requests are responsible for amending their original commits
+following the review process.
+
+There are occasions, however, when a contributor might need to modify patches
+included in pull requests that are submitted by other Zephyr contributors.
+For instance, this is the case when
+
+* a developer picks commits submitted by other contributors into their own
+  pull requests in order to
+
+  * integrate useful content which is part of a stale pull request, or
+  * get content merged to the project's main branch as part of a larger
+    patch
+
+* a developer pushes onto a branch or pull request opened by another
+  contributor in order to
+
+  * assist in revising pull requests and get the patches merged
+    to the project's main branch
+  * finalize review in stale pull requests
+
+
+Recommended policies
+####################
+
+A developer who intends to picking and modifying patches sent by
+another contributor should
+
+* clarify in their pull request the reason for doing that, instead
+  of assisisting in getting the patches merged in their original
+  pull request, and
+* invite the original author of the patches in their pull request review.
+
+A developer who intends to force-push onto a branch or pull request of
+another Zephyr contributor should clarify in the pull request the reason
+for pushing and for modifying the existing patches (e.g. stating that it
+is done for assisting in the pull request review when the pull
+request author is not able to do so).
+
+.. note::
+  Developers should try to limit the above practice to pull requests identified
+  as *stale*.
+
+If the original patches are substantially modified, the developer could
+request the original author to acknowledge that the modified patches may
+be merged while having the original sign-off line and author identity.
+If the original author is not responsive the developer should, alternatively,
+be able to submit the modified patches as their *own* work (i.e. with their
+*own* sign-off line and author identity). In the latter case, the developer
+may want to indentify in the commit message(s) the original source the
```

Comment:
s/indentify/identify/

The criteria and process for characterizing an original author as *unresponsive* are not defined.

Also I would propose that if any material from the original is retained the signed-off-by line of the original should be kept, based on [this discussion](https://stackoverflow.com/a/1962112/236848) of the intent behind that statement (chain-of-responsibility).

Reply:
1. Typo is now fixed.
2. unresponsive authors: yeap, this needs to be defined explicitly, but it is not part of this change, so I prefer to address it outside this PR

>Also I would propose that if any material from the original is retained the signed-off-by line of the original should be kept, based on this discussion of the intent behind that statement (chain-of-responsibility).

We discussed this extensively in the Process Forum (participation is open to all, btw). Retaining the original sign-off cannot be required in all cases; this is exactly the problem we are trying to solve: picking and heavily modifying commits of others, while retaining original sign-off. We don't want to do that, unless the original authors ACKed this.